docs/config/@config.md

```diff
@@ -0,0 +1,197 @@
+
+@RangeInt Use
+-------------
+
+You should use `@RangeInt` to limit an `int` config value. It has a couple of properties:
+
+| Property|Type |    Default Value    |
+|       -:|:-   |        :-:          |
+|      min|int  | `Integer.MIN_VALUE` |
+|      max|int  | `Integer.MAX_VALUE` |
+
+### Example
+```java
+@RangeInt(min = 0) 
+public static int thingFE = 50;
+```
+This will produce an `int` config value with the name `thingFE`, and the default value `50` in the config file; it will ensure the value stays between the bounds of `0` and `Integer.MAX_VALUE`.
+
+@RangeDouble Use
+----------------
+
+You should use `@RangeDouble` to limit a `double` config value. It has a couple of properties:
+
+| Property|Type   |    Default Value   |
+|       -:|:-     |        :-:         |
+|      min|double | `Double.MIN_VALUE` |
+|      max|double | `Double.MAX_VALUE` |
+
+### Example
+```java
+@RangeDouble(min = 0, max = Math.PI) 
+public static double chanceToDrop = 2;
+```
+This will produce a `double` config value with the name `chanceToDrop`, and the default value `Math.PI` in the config file; it will ensure the value stays between the bounds of `0` and `Math.PI`.
+
+@LangKey Use
+------------
+
+You should use the `@LangKey` annotation to give a config a user friendly via .lang files in the mod options menu. NOTE: This does not affect anything in the config file itself. This has 1 property:
+| Property | Type    | Default Value |
+|        -:|:-       |      :-:      |
+|     value|`String` |      N/A      | 
+
+### Example
+```java
+@LangKey("modid.config.thingFE")
+public static int thingFE = 50; 
+```
+This will produce an `int` config value with the name `thingFE`, and the default value `50` in the config file. However in the mod options menu, it will have the translated name supplied from the lang file with the `@LangKey`.
+
+@RequiresMcRestart Use
+----------------------
+
+You should include this annotation on a config that if changed, will require the game to be restarted for the change to take affect.
+
+### Example
+```java
+@RequiresMcRestart
+public static boolean overlayEnabled = false;
+```
+This will make the game require a restart if the value is changed in the configs menu.
+
+@RequiresWorldRestart Use
+-------------------------
+
+This will force the game to be restarted if the config is changed in the mod options menu.
+
+### Example
+```java
+@RequiresWorldRestart
+public static boolean someOtherworldlyThing = false;
+```
+This will force the world to be restarted if the config is changed in the mod options menu.
+
+Nested Configs
+--------------
+A Nested config is possible like this:
+```java
+@Config(modid = "modid")
+public class Configs {
+  public static NestedConfig nestedConfig = new NestedConfig();
+  
+  public static class NestedConfig {
+    public String message = "Don't go that way";
+  }
+}
+```
+
+Sub Categories
+--------------
```

Comment:
What exactly is the difference between a Sub Category and a Nested config? They look pretty much the same to me

Reply:
I assumed a difference, which didn't exist...  What an ass that made me look >.>